pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -519,18 +519,21 @@ private SegmentPartitionMetadata getPartitionMetadataFromSegmentMetadata(Segment
 
   }
 
-  protected boolean writeSegmentsToPropertyStore(String oldZnodePath, String newZnodePath, ZNRecord oldRecord, ZNRecord newRecord,
-      final String realtimeTableName, int expectedVersion) {
-
-    boolean success = _propertyStore.set(oldZnodePath, oldRecord, expectedVersion, AccessOption.PERSISTENT);
+  protected boolean writeSegmentToPropertyStore(String znodePath, ZNRecord znRecord, final String realtimeTableName,
+      int expectedVersion) {
+    boolean success = _propertyStore.set(znodePath, znRecord, expectedVersion, AccessOption.PERSISTENT);
     if (!success) {
-      LOGGER.error("Failed to write old segments to property store for table {}. Expected zookeeper version number: {}",
+      LOGGER.error("Failed to write segment to property store for table {}. Expected zookeeper version number: {}",
           realtimeTableName, expectedVersion);
       return false;
     }
-    success = _propertyStore.set(newZnodePath, newRecord, AccessOption.PERSISTENT);
+    return success;
+  }
+
+  protected boolean writeSegmentToPropertyStore(String znodePath, ZNRecord znRecord, final String realtimeTableName) {
```

Comment:
can we call the other method with expectedVerion set to -1.

Reply:
suspect this is for testing so that we can indepdently override each method. will leave it to neha

Reply:
Yes, in the tests the write method checks for an increasing version number, and it will break if I send -1. Will try to see if I can easily modify test to handle this.